pkg/neg/controller.go

```
@@ -209,14 +214,79 @@ func (c *Controller) processService(key string) error {
 	}
 
 	if !enabled {
+		if _, ok := service.Annotations[annotations.NEGStatusKey]; ok {
+			delete(service.Annotations, annotations.NEGStatusKey)
+		}
 		c.manager.StopSyncer(namespace, name)
+		c.serviceLister.Update(service)
 		return nil
 	}
 
 	glog.V(2).Infof("Syncing service %q", key)
-	// Only service ports referenced by ingress are synced for NEG
-	ings := getIngressServicesFromStore(c.ingressLister, service)
-	return c.manager.EnsureSyncers(namespace, name, gatherSerivceTargetPortUsedByIngress(ings, service))
+	// map of ServicePort (int) to TargetPort
+	svcPortMap := make(annotations.PortNameMap)
+
+	if annotations.FromService(service).NEGEnabledForIngress() {
+		// Only service ports referenced by ingress are synced for NEG
+		ings := getIngressServicesFromStore(c.ingressLister, service)
+		svcPortMap = gatherPortMappingUsedByIngress(ings, service)
+	}
+
+	if annotations.FromService(service).NEGExposed() {
+		knownPorts := make(annotations.PortNameMap)
+		for _, sp := range service.Spec.Ports {
+			knownPorts[sp.Port] = sp.TargetPort.String()
+		}
+
+		negSvcPorts, err := annotations.FromService(service).NEGServicePorts(knownPorts)
+		if err != nil {
+			return err
+		}
+
+		svcPortMap = svcPortMap.Union(negSvcPorts)
+	}
+
+	c.syncNegAnnotation(namespace, name, service, svcPortMap)
```

Do not eat error here. return if there is error. That means there is a Service conflict. It will retry